docs/modules/ROOT/pages/user-guide/bootstrap.adoc

```diff
@@ -34,16 +37,68 @@ public class Configurations {
     }
 }
 ----
+<1> Camel uses the uri scheme to look-up component from its registry wich requires to add the `@Named` annotation to the method otherwise the CDI container would create an anonymous bean and Camel would not be able to look it up.  
+
+Because in Camel Quarkus the Camel components are discovered during the Quarkus' augmentation phase, producing a new component as shown in the example above would invalidate any optimization that may have been made.
+
+As a better alternative you can use `@Inject` to obtain an instance of a component automatically created by Camel or you can observe one of the https://github.com/apache/camel-quarkus/tree/master/extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/events[events] fired by Camel Quarkus as shown in the the following examplein in which we use `@Observes` to be notified about components added to the Camel Context:
+
+[source,java]
+----
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import org.apache.camel.quarkus.core.events.ComponentAddEvent;
+import org.apache.camel.component.log.LogComponent;
+
+@ApplicationScoped
+public static class EventHandler {
+    public void onComponentAdd(@Observes ComponentAddEvent event) {
+        if (event.getComponent() instanceof LogComponent) {
+            // do something with the log component
+        }
+    }
+}
+----
+
+== Camel Main
 
 To configure components and other aspects of Apache Camel through properties, you can add the `camel-quarkus-main`
 extension which brings functionalities from https://camel.apache.org/components/latest/others/main.html[Apache Camel
 Main] to Camel Quarkus.
 
-In the example below, we perform the same configuration of the Log component via properties like we did above using Java:
+In the example below, we apply the same configuration as the one from the Java example above by using properties:
 
 [source,properties]
 ----
 camel.component.log.exchange-formtatter = #class:org.apache.camel.support.processor.DefaultExchangeFormatter
 camel.component.log.exchange-formtatter.show-exchange-pattern = false
 camel.component.log.exchange-formtatter.show-body-type = false
 ----
+
+In addition to support for configuring Camel through properties, `camel-quarkus-main` allows to use conventions to configure the Camel behavior as example, if there a single `ExchangeFormatter` instance in the CDI container, then it will automatically wire that bean to the `LogComponent`.
```

Comment:
```suggestion
In addition to support configuring Camel through properties, `camel-quarkus-main` allows you to use conventions to configure the Camel behavior. For example, if there is a single `ExchangeFormatter` instance in the CDI container, then it will automatically wire that bean to the `LogComponent`.
```